### ISSUE TITLE
Module name check is failing in pylint 2.17.x

### ISSUE THREAD

**Issue by Wayonb** [pylint-dev/pylint#8518]
### Bug description

There seems to be a regression in pylint 2.17.0 where the module name check failed.
It doesnt seems to fail all the time but one of our python project is failing the module name check most of the time.

Pylint 2.16, there wasnt an issue.

### Configuration

```ini
# Regular expression matching correct function names
function-rgx=(([a-z][a-z0-9_]{2,35})|(_[a-z0-9_]*))$

# Good variable names which should always be accepted, separated by a comma
good-names=i,j,k,ex,Run,_

# Include a hint for the correct naming format with invalid-name
include-naming-hint=no

# Regular expression matching correct inline iteration names
inlinevar-rgx=[A-Za-z_][a-z0-9_]*$

# Regular expression matching correct method names
method-rgx=(([a-z][a-z0-9_]{2,35})|(test_[a-z0-9_]+)|(_[a-z0-9_]*))$

# Regular expression matching correct module names
module-rgx=((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$
```

### Command used

```shell
python3 -m pylint --rcfile /home/wayon/code/symbol/product2/linters/python/.pylintrc --load-plugins pylint_quotes --disable ''
```

### Pylint output

```shell
************* Module _chart_utils
tests/test_chart_utils.py:1:0: C0103: Module name "_chart_utils" doesn't conform to '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' pattern (invalid-name)
************* Module _NetworkConnector
tests/test_NetworkConnector.py:1:0: C0103: Module name "_NetworkConnector" doesn't conform to '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' pattern (invalid-name)
************* Module _VersionChartBuilder
tests/test_VersionChartBuilder.py:1:0: C0103: Module name "_VersionChartBuilder" doesn't conform to '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' pattern (invalid-name)
************* Module _NetworkRepository
tests/test_NetworkRepository.py:1:0: C0103: Module name "_NetworkRepository" doesn't conform to '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' pattern (invalid-name)
************* Module _app
tests/test_app.py:1:0: C0103: Module name "_app" doesn't conform to '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' pattern (invalid-name)
************* Module _RoutesFacade
tests/test_RoutesFacade.py:1:0: C0103: Module name "_RoutesFacade" doesn't conform to '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' pattern (invalid-name)
************* Module _HeightChartBuilder
tests/test_HeightChartBuilder.py:1:0: C0103: Module name "_HeightChartBuilder" doesn't conform to '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' pattern (invalid-name)

-------------------------------------------------------------------
Your code has been rated at 9.97/10 (previous run: 10.00/10, -0.03)

https://jenkins.symboldev.com/blue/organizations/jenkins/Symbol%2Fgenerated%2Fproduct%2Fnodewatch/detail/PR-343/10/pipeline/66/
```

### Expected behavior

1. I would not expect to get the failure above.
2. Even if it does fail, the module name above from pylint does not match the filenames.
https://github.com/symbol/product/tree/dev/explorer/nodewatch/tests

Example:  The test file is [test_HeightChartBuilder.py](https://github.com/symbol/product/blob/dev/explorer/nodewatch/tests/test_HeightChartBuilder.py) but pylint only shows it as ``_HeightChartBuilder``

### Pylint version

```shell
isort==5.12.0
pycodestyle==2.10.0
pylint==2.17.1
pylint-quotes==0.2.3
```

### OS / Environment

ubuntu

### Additional dependencies

python --version
Python 3.10.6

**Comment by DanielNoord** [comment]
I can't reproduce:
```python
❯ pylint --module-rgx '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' '/Users/daniel/DocumentenLaptop/Programming/Github/pylint/product/explorer/nodewatch/tests/test_VersionChartBuilder.py'

------------------------------------
Your code has been rated at 10.00/10
```

Looking at your setup it is very likely that something is going wrong with the symlink to the `pylintrc` file and that `pylint` just uses the default on some of the runs. The inconsistent results per run also make me feel that way.

**Comment by Wayonb** [comment]
@DanielNoord Thanks for taking a look at this issue.

Why would you think that pylint is not able to find the ``pylintrc`` when the error actually have the correct ``module-rgx`` expression in the error output?  Shouldnt it show the default ``module-rgx`` value?

**Comment by DanielNoord** [comment]
Hm that is indeed strange, but the fact that this doesn't happen every time and the fact that I can't reproduce still leads me to believe it is something in your setup. We don't strip file names so that is something that I find quite strange about your output.

**Comment by Wayonb** [comment]
@DanielNoord Thanks.

I did some more digging into the files that are passed to pylint.
We are passing a list of files to pylint and file names looks good.
It still does not fail on every run but if I pass in one file it always succeed.
This issue is with passing a list of files to pylint

```
 ~/code/symbol/product2/explorer/nodewatch$ FILES=$(find . -name "*.py")

~/code/symbol/product2/explorer/nodewatch$ echo $FILES
./nodewatch/RoutesFacade.py ./nodewatch/HeightChartBuilder.py ./nodewatch/NetworkRepository.py ./nodewatch/NetworkConnector.py ./nodewatch/__init__.py ./nodewatch/VersionChartBuilder.py ./nodewatch/chart_utils.py ./tests/test/CertificateTestUtils.py ./tests/test_chart_utils.py ./tests/model/test_NodeInfo.py ./tests/model/__init__.py ./tests/model/test_Endpoint.py ./tests/model/test_PacketHeader.py ./tests/test_NetworkConnector.py ./tests/test_VersionChartBuilder.py ./tests/test_NetworkRepository.py ./tests/test_app.py ./tests/test_RoutesFacade.py ./tests/__init__.py ./tests/test_HeightChartBuilder.py ./tests/client/test_CertificateUtils.py ./tests/client/test_CatapultCertificateProcessor.py ./tests/client/test_BasicClient.py ./tests/client/test_SymbolPeerClient.py ./tests/client/test_SymbolClient.py ./tests/client/__init__.py ./tests/client/test_NemClient.py ./puller/model/Exceptions.py ./puller/model/NodeInfo.py ./puller/model/PacketHeader.py ./puller/model/__init__.py ./puller/model/Endpoint.py ./puller/__init__.py ./puller/bindings/__init__.py ./puller/bindings/openssl.py ./puller/client/CertificateUtils.py ./puller/client/SymbolClient.py ./puller/client/SymbolPeerClient.py ./puller/client/CatapultCertificateProcessor.py ./puller/client/__init__.py ./puller/client/NemClient.py ./puller/client/BasicClient.py ./cffi_src/__init__.py ./cffi_src/openssl_build.py

~/code/symbol/product2/explorer/nodewatch$ python3 -m pylint  --rcfile /home/wayon/code/symbol/product2/linters/python/.pylintrc   --load-plugins pylint_quotes    --disable "${PYLINT_DISABLE_COMMANDS}" ${FILES}
************* Module _chart_utils
tests/test_chart_utils.py:1:0: C0103: Module name "_chart_utils" doesn't conform to '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' pattern (invalid-name)
************* Module _NetworkConnector
tests/test_NetworkConnector.py:1:0: C0103: Module name "_NetworkConnector" doesn't conform to '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' pattern (invalid-name)
************* Module _VersionChartBuilder
tests/test_VersionChartBuilder.py:1:0: C0103: Module name "_VersionChartBuilder" doesn't conform to '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' pattern (invalid-name)
************* Module _NetworkRepository
tests/test_NetworkRepository.py:1:0: C0103: Module name "_NetworkRepository" doesn't conform to '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' pattern (invalid-name)
************* Module _app
tests/test_app.py:1:0: C0103: Module name "_app" doesn't conform to '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' pattern (invalid-name)
************* Module _RoutesFacade
tests/test_RoutesFacade.py:1:0: C0103: Module name "_RoutesFacade" doesn't conform to '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' pattern (invalid-name)
************* Module _HeightChartBuilder
tests/test_HeightChartBuilder.py:1:0: C0103: Module name "_HeightChartBuilder" doesn't conform to '((test_)?([A-Za-z][a-zA-Z0-9_]+)|(__main__))$' pattern (invalid-name)

------------------------------------------------------------------
Your code has been rated at 9.97/10 (previous run: 9.97/10, +0.00)
```

**Comment by Wayonb** [comment]
@DanielNoord is the above supported?  Or does pylint only support a list of modules and not a list of files?

**Comment by Wayonb** [comment]
FYI: @DanielNoord I found the issue.  It seems we were missing a ``__init__.py`` in a test folder.  I after I add one it seems to always work.

**Comment by DanielNoord** [comment]
Hmm that is indeed likely to be a cause of issues. Although it is somewhat supported to lint individual files, the order in which they are passed can determine the behaviour of `pylint`. It's weird that this causes the `invalid-name` check to flag but since this isn't reproducible I don't think we can act upon this.

Good to hear it was solved though!
